pkg/shp/flags/build.go

```diff
@@ -14,44 +13,41 @@ func pointerUInt(value uint) *uint {
 }
 
 // BuildSpecFromFlags creates a BuildSpec instance based on command-line flags.
-func BuildSpecFromFlags(flags *pflag.FlagSet) *buildv1alpha1.BuildSpec {
-	clusterBuildStrategyKind := buildv1alpha1.ClusterBuildStrategyKind
-	bundlePruneOption := buildv1alpha1.PruneNever
-	spec := &buildv1alpha1.BuildSpec{
-		Source: buildv1alpha1.Source{
-			Credentials:     &corev1.LocalObjectReference{},
-			Revision:        ptr.To(""),
-			ContextDir:      ptr.To(""),
-			URL:             ptr.To(""),
-			BundleContainer: &buildv1alpha1.BundleContainer{Prune: &bundlePruneOption},
+func BuildSpecFromFlags(flags *pflag.FlagSet) (*buildv1beta1.BuildSpec, *string, *string) {
+	clusterBuildStrategyKind := buildv1beta1.ClusterBuildStrategyKind
+	pruneOption := buildv1beta1.PruneNever
+	spec := &buildv1beta1.BuildSpec{
+		Source: &buildv1beta1.Source{
+			ContextDir: ptr.To(""),
+			Git: &buildv1beta1.Git{
+				Revision:    ptr.To(""),
+				CloneSecret: ptr.To(""),
+			},
+			OCIArtifact: &buildv1beta1.OCIArtifact{
+				Prune:      &pruneOption,
+				PullSecret: ptr.To(""),
```

Comment:
Seems a bit a weird to me to initialize everything with pointers to empty strings just to be later updated by flag values, if I understand it correctly. Is there any way we can simplify this? What do you think?